src/Chapter1.hs

```diff
@@ -541,8 +543,18 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
+mid :: Int -> Int -> Int -> Int
+mid x y z
+ | (x > y) && (y > z) = y
+ | (z > y) && (y > x) = y
+ | (y > x) && (x > z) = x
+ | (z > x) && (x > y) = x
+ | (y > z) && (z > x) = z
+ | (x > z) && (z > y) = z
+ | x == y = x
+ | y == z = y
+ | x == z = x
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you will need to add another guard – `| otherwise = z`, to tell the compiler, that your pattern matching is exhaustive 🙂

It looks like that if you use `<=` and `>=` instead of `<` and `>` in the first cases, you can move all last cases under a single otherwise and return z in here.
The implementation will be a bit shorter 🙂

Reply:
- Thank you for the suggestion. I was not sure about, I don't know for what reason, if haskell knows "<=" and if I know what haskell expects in place of "<="
- Will keep the exhaustion thing in mind, thank you.